<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GenerationTextureInterface</class>
 <widget class="QWidget" name="GenerationTextureInterface">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>764</width>
    <height>565</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Генерация текстуры</string>
  </property>
  <widget class="QLabel" name="titleLabel">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>10</y>
     <width>249</width>
     <height>36</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>16</pointsize>
    </font>
   </property>
   <property name="text">
    <string>Генерация текстуры</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignHCenter</set>
   </property>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>15</x>
     <y>64</y>
     <width>301</width>
     <height>500</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="textureLayout">
        <item>
         <widget class="QLabel" name="textureLabel">
          <property name="text">
           <string>Текстура:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="textureTypeComboBox">
          <property name="currentText">
           <string>Дерево</string>
          </property>
          <item>
           <property name="text">
            <string>Дерево</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="algorithmLayout">
        <item>
         <widget class="QLabel" name="algorithmLabel">
          <property name="text">
           <string>Алгоритм:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="evolutionAlgorithmComboBox">
          <item>
           <property name="text">
            <string>NSGA-III</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>MOEA/D-DE</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>MA-ES</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>LMEA</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>SparseEA</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>PM-MOEA</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QGroupBox" name="generationSettingsGroupBox">
        <property name="title">
         <string>Настройки:</string>
        </property>
        <layout class="QFormLayout" name="generationSettingsLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="mutationProbabilityLabel">
           <property name="text">
            <string>Вероятность мутации:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="mutationProbabilitySpinBox">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="crossoverProbabilityLabel">
           <property name="text">
            <string>Вероятность кроссовера:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="crossoverProbabilitySpinBox">
           <property name="maximum">
            <double>1.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="populationSizeLabel">
           <property name="text">
            <string>Размер популяции:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="populationSizeSpinBox">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>10000</number>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="generationsLabel">
           <property name="text">
            <string>Количество поколений:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QSpinBox" name="generationsSpinBox">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>10000</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="generateButton">
        <property name="text">
         <string>Сгенерировать текстуру</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLabel" name="progressLabel">
        <property name="text">
         <string>Прогресс генерации:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QProgressBar" name="progressBar"/>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="resultLabel">
        <property name="text">
         <string>Результат:</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="textureResultLayout">
        <item>
         <widget class="QLabel" name="textureImageLabel">
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>100</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>100</height>
           </size>
          </property>
          <property name="scaledContents">
           <bool>true</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QPushButton" name="saveButton">
      <property name="text">
       <string>Сохранить текстуру</string>
      </property>
     </widget>
    </item>
    <item>
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QCustomPlot" name="graphicsView" native="true">
   <property name="geometry">
    <rect>
     <x>350</x>
     <y>90</y>
     <width>391</width>
     <height>271</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="qualityChartLabel">
   <property name="geometry">
    <rect>
     <x>410</x>
     <y>60</y>
     <width>271</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>График изменения качества текстур</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
